src/buf_encoder.rs

```diff
@@ -19,6 +19,25 @@ use super::{Case, Table};
 /// This is an important building block for fast hex-encoding. Because string writing tools
 /// provided by `core::fmt` involve dynamic dispatch and don't allow reserving capacity in strings
 /// buffering the hex and then formatting it is significantly faster.
+///
+/// The buffer has a fixed capacity specified when created. The capacity must be an even number since
+/// each byte is encoded as two hex characters.
+///
+/// # Examples
+/// ```
+/// # use hex_conservative::buf_encoder::BufEncoder;
+/// # use hex_conservative::Case;
+/// let mut encoder = BufEncoder::<4>::new(Case::Lower);
+/// encoder.put_byte(0xab);
+/// assert_eq!(encoder.as_str(), "ab");
+/// ```
+/// ```compile_fail
+/// # use hex_conservative::buf_encoder::BufEncoder;
+/// # use hex_conservative::Case;
+/// let mut encoder = BufEncoder::<3>::new(Case::Lower);  // Fails to compile due to odd capacity
+/// encoder.put_byte(0xab);
+/// assert_eq!(encoder.as_str(), "ab");
```

Comment:
Also, if the point of the example is to show that the one line doesn't compile I'd rather replace the lines following `let mut encoder` with `# let _ = encoder;`

Reply:
Done